### ISSUE TITLE
[BUG] Local workspace dependencies installed from registry (instead of symlinked)

### ISSUE THREAD

**Issue by protyposis** [npm/cli#2186]
This bug has been introduced in 7.0.9:

Given the following workspace structure and the assumption that the packages with v1.0.0 are published to the registry:
```
root
 - package.json
 - package-lock.json
 + packages
  + package-a
   - package.json (1.0.0)    
  + package-b
   - package.json (1.0.0)
    -> package-a@1.0.0 (dependency)
```
Expected package directory structure after `npm install|ci` in `root` (npm@7.0.8):

```
root
 + node_modules
  - package-a -> root/packages/package-a (symlink)
 + packages
  + package-a
  + package-b
```

Actual directory structure with npm@7.0.9+:
```
root
 + node_modules
  - package-a -> root/packages/package-a (symlink - correct!)
 + packages
  + package-a
  + package-b
   + node_modules
    - package-a (installed from registry - unexpected!)
```
This leads to the issue that `package-b` uses the actual `package-a@1.0.0` from the registry instead of the local `package-a`. This of course prevents local development.

Workaround: delete unexpected packages from `packages/*/node_modules` folders to force usage of the symlinked packaged from root.

ps: The workspace feature turned out to be extremely buggy and far from being productively usable. What's the reason? Is any of you developers testing or using this feature at all?

**Comment by protyposis** [comment]
Reproducible in 7.0.15.

**Comment by protyposis** [comment]
Seems to be fixed in 7.1.1, thanks!
